### PR TITLE
#32108 - allowOverlappingSegments

### PR DESCRIPTION
**Pull request by daniela-mateescu**
+ remove unused timeline.prop.itemStyle
+ fix a bug (for overlapping rowLayers there was a wrong logic for displaying one beneath each other probably caused by the fact that the logic for displaying the rowLayers was copy pasted from the logic for displaying the segments i.e. itemUtils.rowLayerRenderer was probably copied from itemUtils.rowItemsRenderer)

## Proposed Change:
_What is your change_

## Change type

_Put an `x` in the boxes that apply_

- [ ] Bugfix
- [ ] New feature

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: demo-app/src/stories/allowOverlappingSegments/AllowOverlappingSegments.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+export default {
+    title: "Features/Allow Overlapping Segments"
+}
+
+export const Main = () => {
+    const tasks: Item[] = [
+        ...someTasks,
+        { key: 11, row: 1, title: 'Task AR4', start: d('2018-09-20 10:00'), end: d('2018-09-20 12:00') },
+        { key: 12, row: 2, title: 'Task MD6', start: d('2018-09-20 09:00'), end: d('2018-09-20 15:00') }
+      ];
```
Comment (cristian-spiescu) [inline demo-app/src/stories/allowOverlappingSegments/AllowOverlappingSegments.stories.tsx:17]: Daca am pune aceste 2 noi linii in locul central, ar strica vreun test existent? Daca e ok, hai sa le punem acolo. E ca si politica de la testele CSV: decat sa bagam in testul individual date "ticluite", mai bine le punem in "BD mare".

File: src/timeline.js [hunk 2]
```diff
@@ -462,7 +462,18 @@ export default class Timeline extends React.Component {
...
+    /**
+     * This property controls if the overlapping segments are displayed overlapped
+     * or if they are displayed one beneath the other.
+     *
+     * In this last case, the row expands its height to include all its child segments
+     *
+     * @default false
+     * @type { boolean }
+     */
+    allowOverlappingSegments: PropTypes.bool
```
Comment (cristian-spiescu) [inline src/timeline.js:476]: Momentan, e alba sau neagra. Ar trebui sa poata fi si gri. Adica poate vreau sa activez doar pe anumite tipuri de segmente. 

Vin cu urmatoarea propunere. 
* inlaturam aceasta setare globala
* as pune in `ItemRenderer.allowOverlapping`. Ar fi usor sa te legi de ea? Sa obtii rendererul si sa faci treaba? Daca e greu, o alta varianta ar fi ca in locul prop asta, sa avem o functie, care sa primeasca ceva parametru. Ar fi deci posibila customizare. Insa ar fi un mecanism diferit fata de celelalte, unde lucram la nivel de renderer.
* in 99% din cazuri va fi folosita in modul global. Adica asa cum era variabila de mai sus. Asta o putem face folosind `itemRendererDefaultProps`. Corect?
* ca orice proprietate a ItemRenderer, e duala. E si in props si functie. Functie daca cumva are omul un ItemRenderer custom, si are acolo logica care decide. OK?
* am uitat: la noi in ItemRenderer, daca dam in model o proprietate care e in renderer, e copiata automat? Adica ar putea omul sa aiba tot asa o logica selectiva, dar atunci cand genereaza datele?

Comment (cristian-spiescu) [reply src/timeline.js:476]: UPDATE cf. discutie:

`displayItemOnSeparateRowsIfOverlap: boolean | (item) => boolean`

Daca avem la indemana grup: dam si grup in semnatura: group, item

**Comment by daniela-mateescu** [comment]
Included in https://github.com/flower-platform/react-timeline-10000/pull/72

**Pull request closed without merging**